tests/unit/Core/Checkout/Document/Renderer/DeliveryNoteRendererTest.php

```diff
@@ -0,0 +1,116 @@
+use Shopware\Core\Checkout\Order\OrderDefinition;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\EntitySearchResult;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\Locale\LocaleEntity;
+use Shopware\Core\System\NumberRange\ValueGenerator\NumberRangeValueGeneratorInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * @internal
+ */
+#[Package('after-sales')]
+#[CoversClass(DeliveryNoteRenderer::class)]
+class DeliveryNoteRendererTest extends TestCase
+{
+    public function testRenderCreatesNewOrderVersion(): void
+    {
+        $context = Context::createDefaultContext();
+
+        $documentConfigLoaderMock = new DocumentConfigLoader(
+            $this->createMock(EntityRepository::class),
+            $this->createMock(EntityRepository::class)
+        );
+
+        $order = $this->createOrder();
+        $orderId = $order->getId();
+        $orderSearchResult = new EntitySearchResult(
+            OrderDefinition::ENTITY_NAME,
+            1,
+            new OrderCollection([$order]),
+            null,
+            new Criteria(),
+            $context
+        );
+
+        $orderRepositoryMock = $this->createMock(EntityRepository::class);
+        $orderRepositoryMock
+            ->expects($this->once())
+            ->method('search')
+            ->willReturn($orderSearchResult);
+
+        $orderRepositoryMock
+            ->expects($this->once())
+            ->method('createVersion')
+            ->willReturn('new-order-version-id');
+
+        $connectionMock = $this->createMock(Connection::class);
+        $connectionMock
+            ->expects($this->once())
+            ->method('fetchAllAssociative')
+            ->willReturn([
+                [
+                    'language_id' => Defaults::LANGUAGE_SYSTEM,
+                    'ids' => $orderId,
+                ],
+            ]);
+
+        $deliveryNoteRenderer = new DeliveryNoteRenderer(
+            $orderRepositoryMock,
+            $documentConfigLoaderMock,
+            $this->createMock(EventDispatcherInterface::class),
+            $this->createMock(NumberRangeValueGeneratorInterface::class),
+            $connectionMock,
+            $this->createMock(DocumentFileRendererRegistry::class),
+        );
+
+        $operations = [
+            $orderId => new DocumentGenerateOperation(
+                $orderId
+            ),
+        ];
+
+        $result = $deliveryNoteRenderer->render($operations, $context, new DocumentRendererConfig());
+
+        static::assertArrayHasKey($orderId, $result->getSuccess());
+        static::assertCount(0, $result->getErrors());
```

Comment:
Shouldn't this include an assert to check if the `order_version_id` matches `new-order-version-id`? That is at least what I would expect from a test named `testRenderCreatesNewOrderVersion` 🤔